yurllib/aasa.go

```diff
@@ -276,3 +282,135 @@ func verifyBundleIdentifierIsPresent(content aasaFile, bundleIdentifier string,
 
 	return false
 }
+
+type appleCDNDebugHeaders struct {
+
+	cdnPath string
+	directFileBody string
+	cachedFileBody string
+	cdnStatusCode int
+
+	CDNQuerySuccess bool `json:"CDNQuerySuccess"`
+	NoAppleFailureMessage bool `json:"AppleFailure"`
+	ASACacheUpToDate bool `json:"ASACacheUpToDate"`
+
+	Failure bool `json:"Failure"`
+
+	AppleFailureDetails string `json:"Apple-Failure-Details"` 
+	AppleFailureReason string `json:"Apple-Failure-Reason"` 
+	AppleFrom string `json:"Apple-From"` 
+	AppleTryDirect string `json:"Apple-Try-Direct"` 
+	CacheControl string `json:"Cache-Control"` 
+	ContentLength string `json:"Content-Length"` 
+	Date string `json:"Date"` 
+	Expires string `json:"Expires"` 
+	Via string `json:"Via"` 
+	Age string `json:"Age"` 
+}
+
+
+func readAppleCDNDebugHeaders(cdnResp *http.Response,cdnPath string, directFileBody string) (appleCDNDebugHeaders , error) {
+	result, err := ioutil.ReadAll(cdnResp.Body)
```

Comment:
Looks like you are using err in line 327. Have you tested if result has contents if there is an error?

Reply:
Added if statement to only use result when no err reported